packages/react/src/evaluation/use-feature-flag.ts

```diff
@@ -298,17 +297,30 @@ function attachHandlersAndResolve<T extends FlagValue>(
     suspendUntilReconciled(client);
   }
 
-  const [evaluationDetails, setEvaluationDetails] = useState<EvaluationDetails<T>>(
+  const [evaluationDetails, setEvaluationDetails] = useState<EvaluationDetails<T>>(() =>
```

Comment:
I don't see a reason for this, am I missing something?

Reply:
This prevents us from reevaluating and throwing away the data. Prior to the change, we always ran the evaluation but useState only kept the value from the first run.

Reply:
It's not the fix but an optimization. I'll add a comment next to the fix.

Reply:
BTW, you could perhaps add a test for this optimization (assert the number of evaluations in the provider in some test) but it's not a blocker for me.